lib/vtls/vtls.c

```diff
@@ -460,6 +460,47 @@ bool Curl_ssl_getsessionid(struct Curl_easy *data,
     }
   }
 
+  if(no_match && data->set.ssl.fsslcache && Curl_ssl->load_sessionid) {
+    /*
+     * Search again, this time for a less complete entry, as loaded
+     * from storage.
+     */
```

Comment:
Why check again? Why aren't loaded session ids stored and checked for like normal ones?

Reply:
Loaded session ids are stored like normal ones. They just have less data: no `conn_to_host`, no `conn_to_port`, no valid data in `ssl_config`.
The reason for that was that they are not relevant for the actual `sessionid`, and might even differ between app runs (not sure about that though).
The reason for another for() loop was that the `if()` in the initial one contained an already complicated set of conditions. Adding more to that seemed like it'd have made it way more complicated to review. But of course this can be addressed / changed / updated.